server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -54,7 +58,18 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+@SuppressForbidden(reason = "feature flag overrides")
 public class SearchPipelineServiceTests extends OpenSearchTestCase {
+    @BeforeClass
+    public static void enableFeature() {
```

Comment:
The Gradle runs test in parallel but the system properties (changed from tests) introduce instability since they affect the whole forked JVM that could run a few test suites simultaneously. Could we refactor the test case to use settings instead? Thank you.

Reply:
Do you know if the parallel tests run in their own classloader?

Looking at how we've implemented FeatureFlags, it looks like the settings are held in a static variable, so using `FeatureFlags.initializeSettings` would similarly pollute things across all tests running in the same classloader.

Reply:
For testing purposes (since it's just this test), I think I might add a boolean field to SearchPipelineService that bypasses the FeatureFlags for that specific instance.

Reply:
> Do you know if the parallel tests run in their own classloader?

I think we definitely for process 

> For testing purposes (since it's just this test), I think I might add a boolean field to SearchPipelineService that bypasses the FeatureFlags for that specific instance.

Probably cleaner (and safer) so feature flag won't spread around, thank you